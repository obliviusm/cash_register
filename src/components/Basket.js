import React from 'react';

const showProductCodes = (productCodes) => productCodes.join(", ");

const Basket = ({basket, isLoaded, errorMessage}) => (
  <table>
    <tbody>
      <tr>
        <td>{showProductCodes(basket.productCodes)}</td>
        <td>{basket.price}</td>
      </tr>
    </tbody>
  </table>
);

export default Basket;
