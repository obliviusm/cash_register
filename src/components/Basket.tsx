import React from 'react';
import { useBasket } from '../hooks/useBasket';

const Basket: React.FC = () => {
  const { basket, isLoaded, error } = useBasket();

  if (!isLoaded) {
    return <div>Loading basket...</div>;
  }

  if (error) {
    return <div>Error: {error}</div>;
  }

  return (
    <div>
      <h2>Basket</h2>
      <p>Total Price: ${basket.price}</p>
      <ul>
        {basket.productCodes.map(code => (
          <li key={code}>{code}</li>
        ))}
      </ul>
    </div>
  );
};

export default Basket;
