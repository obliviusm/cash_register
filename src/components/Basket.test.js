import React from 'react';
import { mount } from 'enzyme'
import Basket from './Basket';

it('renders basket', () => {
  const basket = {
    productCodes: ["GR1", "FS1"],
    price: '10.15 â‚¬'
  }
  const component = mount(<Basket basket={basket} error="" isLoaded={true} />)
  expect(component).toMatchSnapshot()
});
