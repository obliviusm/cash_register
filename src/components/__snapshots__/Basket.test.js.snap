// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders basket 1`] = `ReactWrapper {}`;
